toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/GeoViewScope.kt

```diff
@@ -0,0 +1,499 @@
+package com.arcgismaps.toolkit.geoviewcompose
+
+import android.util.DisplayMetrics
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.BoxScope
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.sizeIn
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.drawscope.Fill
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.layout.SubcomposeLayout
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.DpSize
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.times
+import com.arcgismaps.geometry.AngularUnit
+import com.arcgismaps.geometry.Point
+import com.arcgismaps.mapping.ViewpointType
+import com.arcgismaps.mapping.view.DoubleXY
+import com.arcgismaps.mapping.view.DrawStatus
+import com.arcgismaps.mapping.view.GeoView
+import com.arcgismaps.mapping.view.MapView
+import com.arcgismaps.mapping.view.SceneLocationVisibility
+import com.arcgismaps.mapping.view.SceneView
+import com.arcgismaps.mapping.view.ScreenCoordinate
+import com.arcgismaps.mapping.view.zero
+import kotlinx.coroutines.flow.transformWhile
+import kotlin.math.cos
+import kotlin.math.roundToInt
+import kotlin.math.sin
+
+/**
+ * The receiver class of the MapView/SceneView content lambda.
+ *
+ * @since 200.5.0
+ *
+ */
+public sealed class GeoViewScope(private var _geoView: GeoView?, className: String) {
```

Comment:
Better mark the constructor as protected. Why is `geoView` nullable and mutable?
```suggestion
public sealed class GeoViewScope protected constructor(private val geoView: GeoView, className: String) {
```

Reply:
The constructor of sealed class is protected by default, so didn't specify it explicitly here. If you still think we should be explicit I can change it. 
_Constructors of sealed classes can have one of two [visibilities](https://kotlinlang.org/docs/visibility-modifiers.html): protected (by default) or private:_ 

I don't think it has to be mutable, I have updated it. We made it nullable so we can throw the geoview not initialized error if the user tries to use the callout outside of the geoview.

Reply:
Updated.